File: classworks/ege/27ege/data/142/27-142_bad.pas
##
function dist( i, j, K: integer ) :=
   min( abs(j-i), K-abs(j-i) );

Assign( input, '27-142b.txt' );
var (N, K, M) := ReadInteger3;
var D := 20;
var data := |0|*K;

for var i:=1 to N do begin
  var (p, x):= ReadInteger2;
  data[p mod K] := x > 0 ? (x-1) div D + 1 : 0;
end;     

var (minCost, pos) := (MaxInt, -1);
for var i:=0 to K-1 do begin
  var s := (0..K-1).Select( j-> dist(i,j,K)*data[j] ).Sum;
  if data[i] = 0 then begin
    var sumMM := (i-M+1..i+M-1).Select( j->data[ (j+K) mod K] ).Sum;
    if (sumMM = 0) and (s < minCost)  then begin 
      minCost := s;
      pos := i;
    end;  
  end;
end;      

Print( pos, minCost );

File: classworks/ege/27ege/data/142/27-142_bad.py
with open('27-142a.txt') as F:
  N, K, M = map( int, F.readline().split() )
  D = 20
  data = [0]*K
  for i in range(N):
     p, x = map( int, F.readline().split() )
     data[p % K] = (x-1)//D + 1

def dist( i, j ):
   return min( abs(j-i), K-abs(j-i) )

minCost = 10**10
for i in range(K):
  s = sum( dist(i,j)*data[j] for j in range(K) )
  if not data[i]:
    for j in range(i-M+1,i+M):
      if data[j%K]: break
    else:
      if s < minCost:
        minCost = s
        pos = i
        #print( i, s )

print( pos, minCost )

"""
N, K, M = 7, 20, 2
data = [0]*K
data[1] = 100
data[3] = 134
data[5] = 24
data[8] = 57
data[12] = 89
data[15] = 35
data[20%K] = 435
"""
